src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -202,6 +204,18 @@ public JnlpAgentEndpoint resolve() throws IOException {
                             salURL + " is invalid: " + con.getResponseCode() + " " + con.getResponseMessage()));
                     continue;
                 }
+
+                // Check if current version of agent is supported
+                String minimumSupportedVersionHeader = first(header(con, "X-Remoting-Minimum-Version"));
```

Comment:
Header might be missing while connecting to an older Jenkins version. It needs to be handled here

Reply:
>Header might be missing while connecting to an older Jenkins version. It needs to be handled here

Oops I did forget that bit, thanks for telling me!
> Here I would recommend to consider just copying the relevant code into a restricted utility class

I see, then I'll copy the relevant code as well as testcases to remoting's source
>In some cases Jenkins admins may want to ignore the old version warning, e.g. if they use packaged client versions they cannot easily update. Maybe it makes sense to introduce a system property or a CLI argument which allows to opt-out from the check (so that there is a warning instead of an error)

One interesting thing I found out is that Jenkins master only displays a warning for deprecated agent version ([code](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/slaves/SlaveComputer.java#L630)). Maybe we can do the same on agent side, i.e. warning instead of error. I think that introducing a new property/argument for backward compatibility check sounds a tad cumbersome though. And if we use warning we can have the same standard as master's side.

Please share with me what you think, thank you for your time.